### DISCUSSION TITLE
Can the temporal interval of a "Stasis" be chosen by fiat?

### DISCUSSION THREAD

**Discussion by nklsbckmnn** [CommonCoreOntology/CommonCoreOntologies#342] [Announcements]
"Stasis" is currently defined as "A Process in which some Independent Continuant endures and one or more of the dependent entities it bears does not change in kind or intensity."

This seems to me to be ambiguous between 

(a) a Stasis occuring on some (random) fiat temporal interval during which "some Independent Continuant endures and one or more of the dependent entities it bears does not change in kind or intensity"

and 

(b) a Stasis occuring on some longest continous temporal interval during which "some Independent Continuant endures and one or more of the dependent entities it bears does not change in kind or intensity".

Both interpretations seem to be useful.

Figure 11 from the overview paper seems to suggest interpretation (b).

If (b) is supposed to be the intended interpretation, one could (and maybe should) also assert that a "Stasis" is always "process preceded by" some "Gain" and always "process precedes" some "Loss". Although if "Gain" and "Loss" have temporal extension we might not even want to have them in cases where e. g. a continuant bears a role for the entirety of its (the continuant's) existence ("history").

If the intended interpretation is (a), there should be a universal for (b), maybe "Total Stasis" or "Maximal Stasis".

**Comment by mark-jensen** [comment]
@eliasweatherfield Thanks for the interesting comment. I agree that both interpretations are helpful. I have used Stasis in both ways. In many scenarios, one doesn't know if it is maximal or not, just some associated time frame or duration. When it is known to be the maximal case, then bracketing it with Gain and Loss is potentially a good way to ensure the proper interpretation. But, I am not sure a new term that distinguishes the maximal sense is needed in CCO. If the pattern is in the data the maximal cases can be filtered via query. If it's decided to add the term, seems to be a good candidate for an equivalent class, where membership can be determined by inference:

`MaximalStasis equivalentClass
Stasis and (precedes some Loss..)  and (preceded-by some Gain..)`

However, a user that needs this class immediately can add it themselves to their own extension of CCO. Do you have a particular use-case in mind?

**Comment by nklsbckmnn** [comment]
Thanks for the reply. I agree that a new term is not be needed. My worry was about the ambiguity of the definition. Maybe a comment could be added that explains that non-maximal/fiat stases are allowed.

**Comment by johnbeve** [comment]
I take it the definition of Stasis was - and though it has been changed, still is - intentionally broad, so subclasses could be carved out of it, such as, say, Stasis restricted to some fiat interval (if needed). 

Because this is general modeling guidance that trades on the underlying subclass semantics, rather than add a comment to the Stasis class, this might be better suited for a User Guide?

For SA - The updated definition of Stasis is: 
`A Process in which one or more Independent Continuants endure in an unchanging condition.`
Which exhibits the desirable flexibility mentioned above.

**Comment by cameronmore** [comment]
Talking with @jonathanvajda about this, in light of recent discussions about process profile, if we agree with Barry's interpretation of processes that processes _are_ changes, then it makes no sense to say that there is a process which is defined as not changing.

You could talk about a process that stabilizes some independent continuant, like a person agent_in the stabilizing act of balancing, which is about their unchanging (non BFO) 'quality' of being upright but at the same time doing all sorts of things to stay upright, using their muscles, using their coordination cognitive systems, etc. This seems like a bit of a stretch, though it's a clever solution which may have good uses.

A stasis seems to be, Jon and I discussed, an analogous thing to immaterial entities in the continuant tree. As an object is to a site, so a process is to a stasis. What do I mean? I mean that objects serve as the material boundaries of sites, and stases, unchanging temporal intervals, have _changing_ bounds, namely processes which change the thing which was unchanging. 

Suppose we had something like a Stasis of Peace in Gotham City, which is a state of affairs determined by certain measurements and datapoints (this may be a bad example). That stasis has its end when other processes disrupt it, like an increase in some rate of crime, the corruption in the DA's office, or other factors. The peace was an unchanging duration of time which was bounded by processes. A new instance of that peace may be brought about by Batman fighting crime and ending those processes which serve as bounds for the stasis of peace--suppose he locks up all the criminals and beats all the super villains.

Again, this only holds if we interpret `bfo:processes` as changes themselves, which I am in favor of doing, but I'm open to objections and further discussion. This came about while thinking of process profile, which is an unsettled issue anyway.

**Comment by johnbeve** [comment]
@cameronmore Your response here perhaps warrants a separate issue; I think this one can be closed. 

In any event, you write:  

> it makes no sense to say that there is a process which is defined as not changing

I think you mean it _only_ makes sense to say there is a process defined as not changing. No process changes, but any change is always via some process. We can still describe _difference_ across processes, e.g. 

- John walking at 5mph Process having temporal extent t<sub>1</sub>
- John walking at 6mph Process having temporal extent t<sub>2</sub>
- t<sub>1</sub> immediately precedes t<sub>2</sub>
- ...

> and stases, unchanging temporal intervals, have changing bounds,

Stases are not temporal intervals, but they have temporal interval extent.

> As an object is to a site, so a process is to a stasis.

I take you to mean that: any site is bounded by or part of some material entity, and similarly any stasis is bounded by or part of some process. 

I think talking about _changing_ bounds muddies the waters, but if the preceding is what you have in mind then we don't obviously need to get dirty.

**Comment by cameronmore** [comment]
I'll leave it open until we come down on a way to handle stasis going forward, unless someone else wants to close it. I think your description of what I wrote is accurate, maybe this means making stasis a sibling to process, like an antiprocess (again, if we define process as changes). Happy to discuss the details elsewhere going forward.

**Comment by johnbeve** [comment]
@cameronmore I am closing this since the initiating issue was addressed.

See https://github.com/CommonCoreOntology/CommonCoreOntologies/discussions/340 to continue discussion.
